src/ert/config/external_ert_script.py

```diff
@@ -20,22 +24,39 @@ def __init__(self, ert: EnKFMain, storage: StorageAccessor, executable: str):
         self.__job: Optional[Popen[bytes]] = None
 
     def run(self, *args: Any) -> None:
-        command = [self.__executable]
-        command.extend([str(arg) for arg in args])
+        script_content = (
+            Path(self.__executable).read_text(encoding="utf-8").splitlines()
+        )
+        tmp_script_path: Optional[str] = None
+        if self.__executable[-2:] == "sh":
+            script_content.insert(1, "set -e")
+        try:
+            with tempfile.NamedTemporaryFile(
+                mode="w+", encoding="utf-8", delete=False
+            ) as tmp_file:
+                tmp_file.write("\n".join(script_content))
+                tmp_file.flush()
+                tmp_script_path = tmp_file.name
+            os.chmod(tmp_script_path, os.stat(tmp_script_path).st_mode | stat.S_IEXEC)
+            command = [tmp_script_path]
+            command.extend([str(arg) for arg in args])
```

Comment:
Remove this. We don't need to handle script in a special way.

```suggestion
```

If the intention is to execute the program even though it doesn't have the execute bit set (silly legacy), you can open the file in binary mode, read the first 2 characters and check if they are `b"#!"`. If they are, read the full line and prepend the contents to the command. Note that this is easy because the first two bytes _must_ be `#!` for this to be a valid script that the OS can execute. Ie, it can't be ` #!` or `\n#!` as you've discovered.

Eg, if the file is a Python script:
```py
#!/usr/bin/env python
# code here
```

You can do this by:
```py
import shlex

def get_script_command(path: Path) -> Optional[List[str]]:
    with open(path, "rb") as f:
        if f.read(2) != b"#!":
            return None  # not a script file
        return shlex.split(f.readline().decode())
```